### PR TITLE
REF: ScanQRCode to TSX

### PR DESCRIPTION
**Pull request by marcosrdz**
## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced type safety for the `ScanQRCode` navigation route, improving clarity and flexibility of parameters.
  - Updated `ScanQRCode` component with refined state management and improved error handling for QR code reading.

- **Bug Fixes**
  - Enhanced error alert messages for consistent user experience.

- **Documentation**
  - Improved inline type annotations for better clarity and maintainability.

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: screen/send/ScanQRCode.tsx [hunk 1]
```diff
@@ -302,9 +249,9 @@ const ScanQRCode = () => {
     }
   };
 
-  const dismiss = () => {
+  const dismiss = (): void => {
     if (launchedBy) {
-      navigation.navigate({ name: launchedBy, params: {}, merge: true });
+      navigation.navigate(launchedBy as keyof ScanQRCodeParamsList, {});
```
Comment (Overtorment) [inline screen/send/ScanQRCode.tsx:254]: did you forget `merge` ?

File: screen/send/ScanQRCode.tsx [hunk 2]
```diff
@@ -282,8 +229,8 @@ const ScanQRCode = () => {
           if (response.didCancel) {
             setIsLoading(false);
           } else {
-            const asset = response.assets[0];
-            if (asset.uri) {
+            const asset = response.assets && response.assets?.length > 0 ? response.assets[0] : null;
```
Comment (Overtorment) [inline screen/send/ScanQRCode.tsx:232]: i think you can drop `response.assets &&`  and rewrite to `response?.assets?.length > 0`

File: screen/send/ScanQRCode.tsx [hunk 3]
```diff
@@ -239,41 +188,39 @@ const ScanQRCode = () => {
...
+      onBarScanned && onBarScanned({ data: Buffer.from(hex, 'hex').toString('base64') });
       return;
     } catch (_) {}
 
     if (!isLoading) {
       setIsLoading(true);
       try {
         if (launchedBy) {
-          navigation.navigate({ name: launchedBy, params: {}, merge: true });
+          navigation.navigate(launchedBy as keyof ScanQRCodeParamsList, {});
```
Comment (Overtorment) [inline screen/send/ScanQRCode.tsx:201]: did you forget `merge` ?

**Review by Overtorment — CHANGES_REQUESTED** [review]
conflicts as well

File: screen/send/ScanQRCode.tsx [hunk 4]
```diff
@@ -298,8 +247,8 @@ const ScanQRCode = () => {
           if (response.didCancel) {
             setIsLoading(false);
           } else {
-            const asset = response.assets[0];
-            if (asset.uri) {
+            const asset = response.assets && response.assets?.length > 0 ? response.assets[0] : null;
```
[the remark this answers is not in the document]
Comment (Overtorment) [reply screen/send/ScanQRCode.tsx:250]: i said the same thing

**Pull request closed without merging**